packages/webpack-plugin/src/webpack-bundle-analysis/webpackBundleAnalysisPlugin.ts

```diff
@@ -4,20 +4,39 @@ import {
   normalizePath,
   type BundleAnalysisUploadPlugin,
   type Asset,
+  buildStartHelper,
+  buildEndHelper,
+  writeBundleHelper,
 } from "@codecov/bundler-plugin-core";
 import * as webpack from "webpack";
 
 import { findFilenameFormat } from "./findFileFormat";
 
-const PLUGIN_NAME = "codecov-webpack-bundle-analysis-plugin";
+// @ts-expect-error this value is being replaced by rollup
+const PLUGIN_NAME = __PACKAGE_NAME__ as string;
+// @ts-expect-error this value is being replaced by rollup
+const PLUGIN_VERSION = __PACKAGE_VERSION__ as string;
 
 export const webpackBundleAnalysisPlugin: BundleAnalysisUploadPlugin = ({
   output,
   options: userOptions,
 }) => ({
   version: "1",
   name: PLUGIN_NAME,
-  pluginVersion: "1.0.0",
+  pluginVersion: PLUGIN_VERSION,
+  buildStart: () => {
+    output.plugin = {
+      name: PLUGIN_NAME,
+      version: PLUGIN_VERSION,
+    };
+    buildStartHelper(output);
```

Comment:
should `buildStartHelper` take `PLUGIN_NAME` and `PLUGIN_VERSION` so we don't duplicate the `output.plugin` assignment?